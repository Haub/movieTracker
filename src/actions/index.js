import { getMovies, getUser, addUser } from "../utils";

export const fetchMovies = () => {
  return async dispatch => {
    dispatch(contentStatus('loading'))
    try {
      const response = await getMovies()
      dispatch(addMovies(response))
      dispatch(contentStatus('resolved'))
    } catch (error) {
      dispatch(contentStatus('error'))
    }
  }
}

export const fetchUser = (name, email, password, avatar) => {
  return async dispatch => {
    dispatch(contentStatus('loading'))
    try {
      let response;
      if (!name) {
        response = await getUser(email, password)
      } else {
        response = await addUser(name, email, password, avatar)
      }
      dispatch(loginUser(response))
      dispatch(contentStatus('resolved'))
    } catch (error) {
      dispatch(contentStatus('error'))
    }
  }
}

export const contentStatus = (string) => ({
  type: 'CONTENT_STATUS',
  status: string
})

export const addMovies = (movies) => ({
  type: "ADD_MOVIES",
  movies
});

export const loginUser = (user) => ({
  type: 'LOGIN_USER',
  user
})