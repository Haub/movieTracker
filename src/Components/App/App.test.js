/* eslint-disable */
import React from 'react';
import { App } from './index.js';
import { mapStateToProps, mapDispatchToProps } from './index.js';
import { shallow } from 'enzyme';
import { fetchMovies } from '../../actions';

describe('APP', () => {
  let wrapper;
  let mockFetch;
  let mockMovies;
  let mockUser;

  beforeEach(() => {
    mockFetch = jest.fn();
    mockMovies = [];
    mockUser = {};
    wrapper = shallow(
      <App
        fetchMovies = {mockFetch}
        movies={mockMovies}
        user={mockUser}
      />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call fetchMovies on componentDidMount', async () =>{
    let mockMovies = {movies:[]};
    let mockFn = jest.fn().mockImplementation(() => (mockMovies));
    window.fetch = jest.fn().mockImplementation(() => {
      return Promise.resolve({
        json: () => Promise.resolve(mockFn)
      });
    });
    wrapper = shallow(
      <App
        fetchMovies={mockFn}
        movies={[]}
        favorites={[]}
      />);
    await wrapper.instance().componentDidMount();
    expect(mockFn).toHaveBeenCalled();
  });

    it('should have default state login set to false and search set to an empty string', () => {
      expect(wrapper.state().login).toEqual(false);
      expect(wrapper.state().search).toEqual('');
    })

    it('should update state when activateLogin is invoked', () => {
      expect(wrapper.state().login).toEqual(false);
      wrapper.instance().activateLogin();
      expect(wrapper.state().login).toEqual(true);
    })   
});


describe('mapStateToProps', () => {
  it('should have access to movies and current user', () => {
    const mockStore = {
      movies: [],
      user: {}
    }
    const expected = {...mockStore};
    const result = mapStateToProps(mockStore);
    expect(result).toEqual(expected);
  });
})

describe('mapDispatchToProps', () => {
  it('should call dispatch when fetchMovies is invoked', () => {
    const mockDispatch = jest.fn();
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.fetchMovies({title: 'Venom'});
    expect(mockDispatch).toHaveBeenCalled();
  });
})