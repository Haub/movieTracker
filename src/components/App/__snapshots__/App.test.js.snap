// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP should match the snapshot 1`] = `
<div
  className="App"
>
  <div
    className="main-view "
  >
    <Connect(Header)
      activateLogin={[Function]}
      searchMovies={[Function]}
      user={Object {}}
    />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      exact={true}
      path="/favorites"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/:id"
      render={[Function]}
    />
  </div>
</div>
`;
